thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -56,12 +56,43 @@ export default Service.extend({
       return;
     }
 
-    // metrics
+    let metricUrnToOffestAndUrn = {};
     missing.forEach(urn => {
-      return this._fetchSlice(urn, requestContext);
+      const metricUrn = toMetricUrn(urn);
+      if (!(metricUrn in metricUrnToOffestAndUrn)) {
+        metricUrnToOffestAndUrn[metricUrn] = [];
+      }
+      metricUrnToOffestAndUrn[metricUrn].push([toAbsoluteUrn(urn, context.compareMode).split(':')[2].toLowerCase(), urn]);
     });
+
+    Object.keys(metricUrnToOffestAndUrn).forEach(
+        metricUrn => {
+          return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        });
+  },
+
+  _fetchRowSlice(metricUrn, context, metricUrnToOffestAndUrn){
+        const range = context.anomalyRange;
```

Comment:
nit: `const [ start, end ] = context.anomalyRange`
then just plug in ${start} and ${end} in the url

Reply:
documentation: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment